### PR TITLE
feat: add secrets config [fixes DXJ-543]

### PR DESCRIPTION
**Pull request by shamsartem**

### CODE REVIEW

File: src/lib/configs/project/provider.ts [hunk 1]
```diff
@@ -442,57 +513,41 @@ export async function initNewProviderConfig({
...
-    getInitConfigOptions(),
-    getDefault(args),
-  )();
-
-  await ensureSecrets(providerConfig);
-  await ensureConfigToml(providerConfig);
-  return providerConfig;
+  return createProviderConfigWithSecretsAndConfigTomls(name, () => {
+    return getConfigInitFunction(getInitConfigOptions(), getDefault(args))();
+  });
```
Comment (akim-bow) [inline src/lib/configs/project/provider.ts:518]:
```suggestion
  return createProviderConfigWithSecretsAndConfigTomls(name, getConfigInitFunction(getInitConfigOptions(), getDefault(args)));
```
Will it work?

File: src/lib/configs/project/provider.ts [hunk 2]
```diff
@@ -442,57 +513,41 @@ export async function initNewProviderConfig({
...
-
-  await ensureSecrets(providerConfig);
-  await ensureConfigToml(providerConfig);
-  return providerConfig;
+  return createProviderConfigWithSecretsAndConfigTomls(name, () => {
+    return getReadonlyConfigInitFunction(
+      getInitConfigOptions(),
+      getDefault(args),
+    )();
+  });
```
Comment (akim-bow) [inline src/lib/configs/project/provider.ts:530]: same

File: src/lib/configs/project/workers.ts [hunk 3]
```diff
@@ -310,8 +311,8 @@ const migrations: Migrations<Config> = [
 
     return {
       version: 1,
-      ...(Object.keys(deals).length === 0 ? {} : { deals }),
-      ...(Object.keys(hosts).length === 0 ? {} : { hosts }),
+      ...(isEmpty(deals) ? {} : { deals }),
+      ...(isEmpty(hosts) ? {} : { hosts }),
```
Comment (akim-bow) [inline src/lib/configs/project/workers.ts:315]: Thanks for remembering :)

File: src/lib/paths.ts [hunk 4]
```diff
@@ -152,7 +153,7 @@ export const ensureProviderConfigDirPath = async (): Promise<string> => {
   return ensureDir(join(projectRootDir, providerConfigName ?? ""));
 };
 
-export const ensureProviderConfigPath = async (): Promise<string> => {
+export async function ensureProviderConfigPath(): Promise<string> {
```
Comment (akim-bow) [inline src/lib/paths.ts:156]: Do you have a convention whether you use func declaration or func expression?

Comment (akim-bow) [reply src/lib/paths.ts:156]: Just curious

File: src/lib/configs/project/provider.ts (hunk 1 above)
Comment (shamsartem) [reply src/lib/configs/project/provider.ts:518]: yeah, it should, didn't notice that aspect after refactoring

**Pull request merged**